packages/esm-ward-app/src/ward-workspace/ward-patient-notes/form/notes-form.test.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { render, screen } from '@testing-library/react';
+import { showSnackbar, useSession } from '@openmrs/esm-framework';
+import { savePatientNote } from './notes-form.resource';
+import PatientNotesForm from './notes-form.component';
+import { emrApiConfigMock, mockPatient, mockSession } from '__mocks__';
+import useEmrConfiguration from '../../../hooks/useEmrConfiguration';
+
+const testProps = {
+  patientUuid: mockPatient.uuid,
+  closeWorkspace: jest.fn(),
+  closeWorkspaceWithSavedChanges: jest.fn(),
+  promptBeforeClosing: jest.fn(),
+  setTitle: jest.fn(),
+  onWorkspaceClose: jest.fn(),
+  setOnCloseCallback: jest.fn(),
+};
+
+const mockSavePatientNote = savePatientNote as jest.Mock;
+const mockedShowSnackbar = jest.mocked(showSnackbar);
+const mockUseSession = useSession as jest.Mock;
+
+jest.mock('@openmrs/esm-framework', () => ({
+  createErrorHandler: jest.fn(),
+  showSnackbar: jest.fn(),
+  translateFrom: jest.fn(),
+  ResponsiveWrapper: jest.fn(({ children }) => <>{children}</>),
+  useSession: jest.fn().mockReturnValue(() => mockSession),
+}));
```

Comment:
As per updates from @denniskigen, we no longer need to mock the @openmrs/esm-framework.
```
const mockedcreateErrorHandler = jest.mocked (createErrorHandler)
const mockedshowSnackbar = jest.mocked (  showSnackbar)
const mockedtranslateFrom = jest.mocked (  translateFrom)
const mockedResponsiveWrapper = jest.mocked (  ResponsiveWrapper)
const mockeduseSession = jest.mocked (  useSession).mockReturnValue(() => mockSession)

```